src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesRepo.scala

```diff
@@ -18,5 +18,15 @@ trait RoutesRepo {
   def selectAll: DatabasePublisher[RouteRow]
   def selectModifiedSince(localDateTime: LocalDateTime): DatabasePublisher[RouteRow]
   def selectByName(name: String): DatabasePublisher[RouteRow]
-  def delete(id: Long): Future[Boolean]
+
+  /**
+   * Marks route as deleted.
+   *
+   * @param   id of the route to be marked as deleted
+   * @param   dateTime optional timestamp that will be written into deleted_at column
+   * @return  future that contains operation success flag
+   */
+  def delete(id: Long, dateTime: Option[LocalDateTime] = None): Future[Boolean]
+  def selectDeletedBefore(dateTime: LocalDateTime): DatabasePublisher[RouteRow]
```

Comment:
It's nice that you added documentation for the delete method. It would be nice to add some for the selectDeleteBefore also, to make sure that the client knows that this refers to the routes which were marked as deleted.